inventory_management_system_api/services/catalogue_category.py

```diff
@@ -140,3 +149,21 @@ def update(
         return self._catalogue_category_repository.update(
             catalogue_category_id, CatalogueCategoryIn(**{**stored_catalogue_category.model_dump(), **update_data})
         )
+
+    def _check_duplicate_catalogue_item_property_names(
+        self, catalogue_item_properties: List[CatalogueItemPropertySchema]
+    ) -> None:
+        """
+        Go through all the catalogue item properties to check for any duplicate names.
+        :param catalogue_item_properties: The supplied catalogue item properties
+        :raises DuplicateCatalogueItemPropertyName: If a duplicate catalogue item property name is found.
+        """
+        logger.info("Checking for duplicate catalogue item property names")
+        seen_catalogue_item_property_names = set()
```

Comment:
The property names needs to be trimmed before being checked. Currently if you have "name" and "name " (with a space at the end) they are two different values.


  
![dup item prop name](https://github.com/ral-facilities/inventory-management-system-api/assets/83226114/ac561ed0-e691-48c4-adac-a52604c18fdd)